situp-plugins/peer-forwarder/src/test/java/com/amazon/situp/plugins/processor/peerforwarder/PeerForwarderTest.java

```diff
@@ -0,0 +1,84 @@
+    private static final Span SPAN_3 = Span.newBuilder()
+            .setTraceId(ByteString.copyFromUtf8("traceId1")).setSpanId(ByteString.copyFromUtf8("spanId3")).build();
+    private static final Span SPAN_4 = Span.newBuilder()
+            .setTraceId(ByteString.copyFromUtf8("traceId2")).setSpanId(ByteString.copyFromUtf8("spanId4")).build();
+    private static final Span SPAN_5 = Span.newBuilder()
+            .setTraceId(ByteString.copyFromUtf8("traceId2")).setSpanId(ByteString.copyFromUtf8("spanId5")).build();
+    private static final Span SPAN_6 = Span.newBuilder()
+            .setTraceId(ByteString.copyFromUtf8("traceId2")).setSpanId(ByteString.copyFromUtf8("spanId6")).build();
+
+    private static final ExportTraceServiceRequest REQUEST_1 = generateRequest(SPAN_1, SPAN_2, SPAN_4);
+    private static final ExportTraceServiceRequest REQUEST_2 = generateRequest(SPAN_3, SPAN_5, SPAN_6);
+
+    private static ExportTraceServiceRequest generateRequest(final Span... spans) {
+        return ExportTraceServiceRequest.newBuilder()
+                .addResourceSpans(ResourceSpans.newBuilder()
+                        .setResource(Resource.newBuilder().build())
+                        .addInstrumentationLibrarySpans(
+                                InstrumentationLibrarySpans.newBuilder()
+                                        .setInstrumentationLibrary(InstrumentationLibrary.newBuilder().build())
+                                        .addAllSpans(Arrays.asList(spans)))
+                        .build()).build();
+    }
+
+    private static ResourceSpans generateResourceSpans(final Span... spans) {
+        return ResourceSpans.newBuilder().setResource(Resource.newBuilder().build()).addInstrumentationLibrarySpans(
+                InstrumentationLibrarySpans.newBuilder()
+                        .setInstrumentationLibrary(InstrumentationLibrary.newBuilder().build())
+                        .addAllSpans(Arrays.asList(spans))).build();
+    }
+
+    @Test
+    public void testLocalIpOnly() {
+        final PeerForwarder testPeerForwarder = generatePeerForwarder(Collections.emptyList());
+        final List<Record<ExportTraceServiceRequest>> exportedRecords =
+                testPeerForwarder.execute(Arrays.asList(new Record<>(REQUEST_1), new Record<>(REQUEST_2)));
+        Assert.assertTrue(exportedRecords.size() >= 3);
+        final List<ResourceSpans> exportedResourceSpans = new ArrayList<>();
+        for (final Record<ExportTraceServiceRequest> record: exportedRecords) {
+            exportedResourceSpans.addAll(record.getData().getResourceSpansList());
+        }
+        final List<ResourceSpans> expectedResourceSpans = Arrays.asList(
+                generateResourceSpans(SPAN_1, SPAN_2),
+                generateResourceSpans(SPAN_3),
+                generateResourceSpans(SPAN_4),
+                generateResourceSpans(SPAN_5, SPAN_6)
```

Comment:
Just curious - why is this the expected outcome?

Reply:
Yeah. Say in the raw input span 1, 2, 4 is in the only resourceSpan of 1st request, but 1, 2 belong to traceId1 while 4 belongs to traceId2. So it is splitted into two resourcespans 1,2 and 4 grouped by traceId. Similarly for 3,5,6.